mmf/datasets/builders/localized_narratives/database.py

```diff
@@ -55,18 +56,42 @@ def load_annotation_db(self, path):
             for line in f:
                 annotation = json.loads(line)
                 loc_narr = LocalizedNarrative(**annotation)
-                data.append(
-                    {
-                        "dataset_id": loc_narr.dataset_id,
-                        "image_id": loc_narr.image_id,
-                        "caption": loc_narr.caption,
-                        "feature_path": self._feature_path(
-                            loc_narr.dataset_id, loc_narr.image_id
-                        ),
-                    }
-                )
+                data_detail = {
+                    "image_id": loc_narr.image_id,
+                    "feature_path": self._feature_path(
+                        loc_narr.dataset_id, loc_narr.image_id
+                    ),
+                }
+                if (
+                    hasattr(self.config, "trace_mode")
+                    and self.config.trace_mode is not None
```

Comment:
```suggestion
self.config.get("trace_mode", None)
```

hasattr is generally discouraged with OmegaConf.